### PR TITLE
ENG-1796 implementation of DataTable on most tables in Appbuilder

### PR DESCRIPTION
**Pull request by jeffgo10**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **jeffgo10**

If they are not already assigned, you can assign the PR to them by writing `/assign @jeffgo10` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/app/App.js
Comment (ichalagashvili) [inline src/ui/app/App.js:387]: have you tested that this did not break existing DND for example for page list moving or widget dnd?

Comment (jeffgo10) [reply src/ui/app/App.js:387]: @ichalagashvili yes this works because all of the DnD components/functions are now imported via @entando/dndtable. This was the reason why at the first place they moved the dndtable at the appbuilder because the provider came from app-builder while the dndtable came from an imported module. putting both provider and dndtable at the same source should make `react-dnd` work fine.

File: test/ui/pages/config/PageConfigPage.test.js
Comment (gidesan) [inline test/ui/pages/config/PageConfigPage.test.js:7]: please remove comments

**Review by gidesan — CHANGES_REQUESTED** [review]
very big work! 
there's a small comment to remove plus a question: why there are less tests now?

File: test/ui/pages/config/PageConfigPage.test.js
Comment (jeffgo10) [reply test/ui/pages/config/PageConfigPage.test.js:7]: > very big work!
> there's a small comment to remove plus a question: why there are less tests now?

@gidesan most tests that are removed are related to table row tests which are mostly not applicable to the new table component. Though if needed, we can change the tests with the use of react testing library for easy tests but this will take a lot of time on producing unit tests.

Comment (gidesan) [reply test/ui/pages/config/PageConfigPage.test.js]: Yeah, please use RTL

File: test/ui/pages/common/PageTree.test.js
Comment (gidesan) [inline test/ui/pages/common/PageTree.test.js:91]: Can we use a different object with the move positions instead of adding them to the Container?

**Review by gidesan — CHANGES_REQUESTED** [review]
only some small changes needed

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1019#pullrequestreview-602719078" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
